tpgtools/resource.go

```diff
@@ -564,6 +570,12 @@ func createResource(schema *openapi.Schema, info *openapi.Info, typeFetcher *Typ
 		}
 	}
 
+	// Determine if a resource has a create method.
+	res.HasCreate, _ = schema.Extension["x-dcl-has-create"].(bool)
```

Comment:
Why is sending this information to the client and having the client need to change behavior better than containing the no-op or update only within the DCL?

Reply:
The change in behavior is limited to no longer sending the `BlockModification` parameter to the apply method. This seems fairly trivial to me and I'm not sure what a solution contained within the DCL would even look like.